packages/features/ee/round-robin/roundRobinReassignment.ts

```diff
@@ -0,0 +1,517 @@
+  newReferencesToCreate = structuredClone(results.referencesToCreate);
+
+  await BookingReferenceRepository.replaceBookingReferences({
+    bookingId,
+    newReferencesToCreate,
+  });
+
+  // Send to new RR host
+  await sendRoundRobinScheduledEmails(evt, [
+    {
+      ...reassignedRRHost,
+      name: reassignedRRHost.name || "",
+      username: reassignedRRHost.username || "",
+      timeFormat: getTimeFormatStringFromUserTimeFormat(reassignedRRHost.timeFormat),
+      language: { translate: reassignedRRHostT, locale: reassignedRRHost.locale || "en" },
+    },
+  ]);
+
+  if (previousRRHost) {
+    // Send to cancelled RR host
+    // First we need to replace the new RR host with the old RR host in the evt object
+    const cancelledRRHostEvt = cloneDeep(evt);
+    cancelledRRHostEvt.title = currentBookingTitle;
+    if (hasOrganizerChanged) {
+      cancelledRRHostEvt.organizer = {
+        name: previousRRHost.name || "",
+        email: previousRRHost.email,
+        language: {
+          locale: previousRRHost.locale || "en",
+          translate: previousRRHostT,
+        },
+        timeZone: previousRRHost.timeZone,
+        timeFormat: getTimeFormatStringFromUserTimeFormat(previousRRHost.timeFormat),
+      };
+    } else if (cancelledRRHostEvt.team) {
+      // Filter out the new RR host from attendees and add the old RR host
+      const newMembersArray = cancelledRRHostEvt.team?.members || [];
+      cancelledRRHostEvt.team.members = newMembersArray.filter(
+        (member) => member.email !== reassignedRRHost.email
+      );
+      cancelledRRHostEvt.team.members.unshift({
+        id: previousRRHost.id,
+        email: previousRRHost.email,
+        name: previousRRHost.name || "",
+        timeZone: previousRRHost.timeZone,
+        language: { translate: previousRRHostT, locale: previousRRHost.locale || "en" },
+      });
+    }
+
+    await sendRoundRobinCancelledEmails(cancelledRRHostEvt, [
+      {
+        ...previousRRHost,
+        name: previousRRHost.name || "",
+        username: previousRRHost.username || "",
+        timeFormat: getTimeFormatStringFromUserTimeFormat(previousRRHost.timeFormat),
+        language: { translate: previousRRHostT, locale: previousRRHost.locale || "en" },
+      },
+    ]);
+  }
+
+  // Handle changing workflows with organizer
+  if (hasOrganizerChanged) {
+    const workflowReminders = await prisma.workflowReminder.findMany({
+      where: {
+        bookingUid: booking.uid,
+        method: WorkflowMethods.EMAIL,
+        workflowStep: {
+          action: WorkflowActions.EMAIL_HOST,
+          workflow: {
+            trigger: {
+              in: [
+                WorkflowTriggerEvents.BEFORE_EVENT,
+                WorkflowTriggerEvents.NEW_EVENT,
+                WorkflowTriggerEvents.AFTER_EVENT,
+              ],
+            },
+          },
+        },
+      },
+      include: {
+        workflowStep: {
+          include: {
+            workflow: true,
```

Comment:
would `workflowSelect` from `getAllWorkflows.ts` work here instead of using `include`?

Reply:
The `workflowSelect` nesting is `workflow -> steps`. This query is `step -> workflow`. I just selected the fields that would be relevant.

Reply:
Sorry missed that. Can we define all fields we need in a select instead of an include? [Never use Includes - use select](https://handbook.cal.com/engineering/best-practices/only-select-data-you-need)

Reply:
Changed this one to `select`. The other `include` statements are needed.